fundingmanager_test.go

```diff
@@ -1087,14 +1109,67 @@ func TestFundingManagerFundingTimeout(t *testing.T) {
 	}
 
 	// Bob should still be waiting for the channel to open.
-	assertNumPendingChannels(t, bob, 1)
+	assertNumPendingChannelsRemains(t, bob, 1)
 
 	bob.mockNotifier.epochChan <- &chainntnfs.BlockEpoch{
 		Height: fundingBroadcastHeight + 288,
 	}
 
 	// Should not be pending anymore.
-	assertNumPendingChannels(t, bob, 0)
+	assertNumPendingChannelsBecomes(t, bob, 0)
+}
+
+// TestFundingManagerFundingNotTimeoutInitiator checks that if the user was
+// the channel initiator, that it does not timeout when the lnd restarts
+func TestFundingManagerFundingNotTimeoutInitiator(t *testing.T) {
+	disableFndgLogger(t)
+
+	alice, bob := setupFundingManagers(t)
+	defer tearDownFundingManagers(t, alice, bob)
+
+	// We will consume the channel updates as we go, so no buffering is needed.
+	updateChan := make(chan *lnrpc.OpenStatusUpdate)
+
+	// Run through the process of opening the channel, up until the funding
+	// transaction is broadcasted.
+	_ = openChannel(t, alice, bob, 500000, 0, 1, updateChan)
+
+	// Alice will at this point be waiting for the funding transaction to be
+	// confirmed, so the channel should be considered pending.
+	pendingChannels, err := alice.fundingMgr.cfg.Wallet.Cfg.Database.FetchPendingChannels()
+	if err != nil {
+		t.Fatalf("unable to fetch pending channels: %v", err)
+	}
+	if len(pendingChannels) != 1 {
+		t.Fatalf("Expected Alice to have 1 pending channel, had  %v",
+			len(pendingChannels))
+	}
+
+	recreateAliceFundingManager(t, alice)
+
+	// After recreating Alice she should still have one pending channel
+	pendingChannels, err = alice.fundingMgr.cfg.Wallet.Cfg.Database.FetchPendingChannels()
+	if err != nil {
+		t.Fatalf("unable to fetch pending channels: %v", err)
+	}
+	if len(pendingChannels) != 1 {
+		t.Fatalf("Expected Alice to have 1 pending channel, had  %v",
+			len(pendingChannels))
+	}
+
+	alice.mockNotifier.epochChan <- &chainntnfs.BlockEpoch{
```

Comment:
First send `fundingBroadcastHeight + 287` check that both Alice and Bob remain pending, send `fundingBroadcastHeight + 288`, check that only Alice remains pending.